zipline/pipeline/loaders/blaze/_core.pyx

```diff
@@ -0,0 +1,651 @@
+                sid_column_ixs,
+                mask,
+                input_array,
+                missing_value,
+                is_missing_nan,
+                array_kind,
+            )
+        else:
+            return _array_for_column_impl[np.float64_t, AsArrayKind](
+                dtype,
+                out_array,
+                size,
+                ts_ixs,
+                asof_ixs,
+                sids,
+                sid_column_ixs,
+                mask,
+                input_array,
+                missing_value,
+                is_missing_value[np.float64_t],
+                array_kind,
+            )
+    elif kind == 'O':
+        return _array_for_column_impl[object, AsArrayKind](
+            dtype,
+            out_array,
+            size,
+            ts_ixs,
+            asof_ixs,
+            sids,
+            sid_column_ixs,
+            mask,
+            input_array,
+            missing_value,
+            is_missing_value[object],
+            array_kind,
+        )
+    elif kind == 'b':
+        return _array_for_column_impl[np.uint8_t, AsArrayKind](
+            dtype,
+            out_array.view('uint8'),
+            size,
+            ts_ixs,
+            asof_ixs,
+            sids,
+            sid_column_ixs,
+            mask,
+            input_array.view('uint8'),
+            int(missing_value),
+            is_missing_value[np.uint8_t],
+            array_kind,
+        )
+    else:
+        raise TypeError('unknown column dtype: %r' % input_array.dtype)
+
+
+cdef arrays_from_rows(DatetimeIndex_t dates,
+                      object data_query_time,
+                      object data_query_tz,
+                      object assets,
+                      np.ndarray[np.int64_t] sids,
+                      object mask,
+                      list columns,
+                      object all_rows,
+                      AsArrayKind array_kind):
+    cdef dict column_ixs = dict(zip(assets, range(len(assets))))
+
+    cdef Py_ssize_t n
+    cdef list ts_dates_list
+    if data_query_time is not None:
+        ts_dates_list = PyList_New(len(dates))
+        for n, dt in enumerate(dates):
+            combined = pd.Timestamp.combine(
+                dt.date(),
+                data_query_time,
+            ).tz_localize(data_query_tz).tz_convert('utc')
+            Py_INCREF(combined)
+            PyList_SET_ITEM(ts_dates_list, n, combined)
+
+        ts_dates = pd.DatetimeIndex(ts_dates_list)
+    else:
+        ts_dates = dates
+
+    cdef np.ndarray[np.int64_t] ts_ixs = ts_dates.searchsorted(
+        all_rows[TS_FIELD_NAME].values,
+        'right',
+    )
+    cdef np.ndarray[np.int64_t] asof_ixs = dates.searchsorted(
+        all_rows[AD_FIELD_NAME].values,
+        'right',
+    )
+
+    cdef tuple out_shape = (len(dates), len(assets))
+    cdef dict out = {}
+    cdef Py_ssize_t size = len(ts_ixs)
+
+    for column in columns:
+        out[column] = array_for_column[AsArrayKind](
```

Comment:
it looks like you can pass kwargs, but something is weird here because it is fused I think. Whatever the reason, it doesn't compile without this.